### PR TITLE
loss calculation + plotting loss and accuracy

### PR DESCRIPTION
**Pull request by hubert-groen**

### CODE REVIEW

File: resnet.py [hunk 1]
```diff
@@ -50,18 +51,23 @@ def train(self, save_dir, read_index, num_epochs=75, batch_size=256, learning_ra
...
-        train_dataset = datasets.CIFAR10('data/cifar', download=True, transform=train_transform)
+        train_dataset = datasets.CIFAR10('data/cifar', download=True, transform=train_transform, train=True)
+        test_dataset = datasets.CIFAR10('data/cifar', download=True, transform=train_transform, train=False)
+        full_dataset = ConcatDataset([train_dataset, test_dataset])
         train_idx = np.load(f'indices/train_idx_{read_index}.npy')
         if teacher_logits_path is not None:
             teacher_logits = np.load(teacher_logits_path)
-            train_dataset.targets = teacher_logits
-        train_subset = Subset(train_dataset, train_idx)
+            test_dataset.targets = teacher_logits
```
Comment (WojciechGierulski) [inline resnet.py:60]: tu chyba full_dataset?

File: resnet.py [hunk 2]
```diff
@@ -109,44 +115,64 @@ def train(self, save_dir, read_index, num_epochs=75, batch_size=256, learning_ra
...
+            transforms.Normalize((0.4914, 0.4822, 0.4465), (0.2023, 0.1994, 0.2010)),
+        ])
 
-        test_dataset = datasets.CIFAR10('data/cifar', download=True, transform=test_transform, train=False)
+        train_dataset = datasets.CIFAR10('data/cifar', download=True, transform=train_transform, train=True)
+        test_dataset = datasets.CIFAR10('data/cifar', download=True, transform=train_transform, train=False)
+        full_dataset = ConcatDataset([train_dataset, test_dataset])
         train_idx = np.load(f'indices/test_idx_{read_index}.npy')
-        train_subset = Subset(test_dataset, train_idx)
+        train_subset = Subset(full_dataset, train_idx)
```
Comment (WojciechGierulski) [inline resnet.py:149]: daj tu pls test_idx nazwę zmiennej

File: resnet.py [hunk 3]
```diff
@@ -109,44 +115,64 @@ def train(self, save_dir, read_index, num_epochs=75, batch_size=256, learning_ra
...
+
         with torch.no_grad():
             for i, data in enumerate(data_loader, 0):
                 images, labels = data
                 images = images.to(self.device)
                 labels = labels.to(self.device)
 
                 outputs = self.net(images)
+                loss = loss_function(outputs, labels)
+                total_loss += loss.item() * images.size(0)
```
Comment (WojciechGierulski) [inline resnet.py:165]: czemu `* images.size(0)`?

File: resnet.py [hunk 4]
```diff
@@ -109,44 +115,64 @@ def train(self, save_dir, read_index, num_epochs=75, batch_size=256, learning_ra
...
+
                 _, predicted = torch.max(outputs, dim=1)
                 total += labels.size(0)
                 correct += (predicted == labels.flatten()).sum().item()
 
+        accuracy = correct / total * 100
+        average_loss = total_loss / total
```
Comment (WojciechGierulski) [inline resnet.py:172]: tutaj raczej trzeba podzielić przez liczbę batchy, a nie próbek treningowych

**Review by WojciechGierulski — CHANGES_REQUESTED** [review]

File: resnet.py (hunk 1 above)
Comment (hubert-groen) [reply resnet.py:60]: Racja. full_dataset.targets powinno być

File: resnet.py (hunk 2 above)
Comment (hubert-groen) [reply resnet.py:149]: Ok

File: resnet.py (hunk 3 above)
Comment (hubert-groen) [reply resnet.py:165]: Usuwam

File: resnet.py (hunk 4 above)
Comment (hubert-groen) [reply resnet.py:172]: OK, robię / batch_size
